zerver/lib/message.py

```diff
@@ -1003,9 +1003,6 @@ def get_raw_unread_data(user_profile: UserProfile) -> RawUnreadMessagesResult:
         .exclude(
             message__recipient_id__in=excluded_recipient_ids,
         )
```

Comment:
I guess the `get_inactive_recipient_ids` part looks at deactivated streams; I think this is a reminder that we may want to adjust the "mark messages as unread" backend endpoint to skip message IDs that structurally should never be marked as unread (e.g. those sent by muted users or sent to deactivated streams), since those are states we don't want to have exist.

@abhijeetbodas2001 FYI for the muting question.

Reply:
I agree.

Reply:
If we're going to allow users to mark a **particular** message or particular set of messages as unread, I think we should not treat muted sender messages specially.

If we're implementing a wider option (which I think we are in V1?) like "mark topic as unread" or "mark all messages below this message as unread", I think we should have the muted messages retain their state (remain as `read`, that is)